AP regression from new release in Loader 0.14.15

I've got [an accessor](https://github.com/KyoriPowered/adventure-platform-fabric/blob/563db62e608038d15d83178a18bc3735b05afb1b/src/accessor/java/net/kyori/adventure/platform/fabric/impl/accessor/api/key/InvalidKeyExceptionAccess.java#L33) targeting [a class in my own API library](https://github.com/KyoriPowered/adventure/blob/a79cef4e49575551c6eb39ad09df1e674e11b6b7/key/src/main/java/net/kyori/adventure/key/InvalidKeyException.java#L39) that is non-obf. This accessor has been compiling and applying fine in past versions, but with this new Mixin version (0.12.3?) I get the following compile error:

```
C:\Users\abby\dev\kyori-text\adventure-platform-fabric\src\accessor\java\net\kyori\adventure\platform\fabric\impl\accessor\api\key\InvalidKeyExceptionAccess.java:34: error: Invalid Factory @Invoker return type, expected null but found net.kyori.adventure.key.InvalidKeyException
  @Invoker(value = "<init>", remap = false)
  ^
```

I have tried adding `remap = false` to avoid the error, but it has had no effect.

--- a/src/ap/java/org/spongepowered/tools/obfuscation/AnnotatedMixinElementHandlerAccessor.java
+++ b/src/ap/java/org/spongepowered/tools/obfuscation/AnnotatedMixinElementHandlerAccessor.java
@@ -46,8 +46,6 @@ import org.spongepowered.tools.obfuscation.mirror.FieldHandle;
 import org.spongepowered.tools.obfuscation.mirror.MethodHandle;
 import org.spongepowered.tools.obfuscation.mirror.TypeHandle;
 import org.spongepowered.tools.obfuscation.mirror.TypeUtils;
-import org.spongepowered.tools.obfuscation.mirror.TypeUtils.Equivalency;
-import org.spongepowered.tools.obfuscation.mirror.TypeUtils.EquivalencyResult;
 
 import com.google.common.base.Strings;
 
@@ -304,19 +302,11 @@ class AnnotatedMixinElementHandlerAccessor extends AnnotatedMixinElementHandler 
     }
 
     private void registerFactoryForTarget(AnnotatedElementInvoker elem, TypeHandle target) {
-        EquivalencyResult equivalency = TypeUtils.isEquivalentType(this.ap.getProcessingEnvironment(), elem.getReturnType(), target.getTypeMirror());
-        if (equivalency.type != Equivalency.EQUIVALENT) {
-            if (equivalency.type == Equivalency.EQUIVALENT_BUT_RAW && equivalency.rawType == 1) {
-                elem.printMessage(this.ap, MessageType.INVOKER_RAW_RETURN_TYPE, "Raw return type for Factory @Invoker", SuppressedBy.RAW_TYPES);
-            } else if (equivalency.type == Equivalency.BOUNDS_MISMATCH) {
-                elem.printMessage(this.ap, MessageType.FACTORY_INVOKER_GENERIC_ARGS, "Invalid Factory @Invoker return type, generic type args of "
-                        + target.getTypeMirror() + " are incompatible with " + elem.getReturnType() + ". " + equivalency);
-                return;
-            } else {
-                elem.printMessage(this.ap, MessageType.FACTORY_INVOKER_RETURN_TYPE, "Invalid Factory @Invoker return type, expected "
-                        + target.getTypeMirror() + " but found " + elem.getReturnType());
-                return;
-            }
+        String returnType = TypeUtils.getTypeName(elem.getReturnType());
+        if (!returnType.equals(target.toString())) {
+            elem.printMessage(this.ap, MessageType.FACTORY_INVOKER_RETURN_TYPE, "Invalid Factory @Invoker return type, expected "
+                    + target + " but found " + returnType);
+            return;
         }
         if (!elem.isStatic()) {
             elem.printMessage(this.ap, MessageType.FACTORY_INVOKER_NONSTATIC, "Factory @Invoker must be static");

